### PR TITLE
Update workflow files to push NuGet to our private feed

### PR DESCRIPTION
**Pull request by vladtaraban**
# Summary
Updated workflow files to allow build when opening a pull request and push NuGet package when it's needed

### CODE REVIEW

**Comment by vladtaraban** [comment]
Hi @prin53 , added workflow files, but I think that I can't set secrets in this repo, could you please give me access to it or set them by yourself?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+name: Publish NuGet packages
+
+on:
+  workflow_dispatch:
+
+jobs:
+  Mac-NuGet-Builds:
+    env:
+      NUGET_AUTH_TOKEN: ${{ secrets.NUGET_API_TOKEN }}
```
Comment (prin53) [inline .github/workflows/ci.yml:9]: I think this one is never used.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      run: msbuild /t:build,pack /p:Configuration=Release Sources/Microcharts.Droid/Microcharts.Droid.csproj
+
+    - name: "Build and pack Microcharts iOS"
+      run: msbuild /t:build,pack /p:Configuration=Release Sources/Microcharts.iOS/Microcharts.iOS.csproj
+
+    - name: "Build and pack Microcharts Meta-package"
+      run: dotnet pack Sources/Microcharts.Metapackage/Microcharts.Metapackage.csproj
+
+    - name: Add GitHub Packages source
+      run: nuget sources add -name github.com -source https://nuget.pkg.github.com/cabvision/index.json -username ${{ secrets.USERNAME }} -password ${{ secrets.NUGET_SOURCE_API_KEY }}
```
Comment (prin53) [inline .github/workflows/ci.yml:49]: You can use GitHub-provided secret `secrets.GITHUB_TOKEN`.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+    - name: Publish packages to nuget.org
+      run: |
+        cd artifacts
+        ls *.nupkg
+        nuget push *.nupkg --source github.com --api-key $NUGET_AUTH_TOKEN --skip-duplicate
```
Comment (prin53) [inline .github/workflows/ci.yml:55]: You can use `secrets.GITHUB_TOKEN` here as well.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    - name: "Build and pack Microcharts iOS"
+      run: msbuild /t:build,pack /p:Configuration=Release Sources/Microcharts.iOS/Microcharts.iOS.csproj
+
+    - name: "Build and pack Microcharts Meta-package"
+      run: dotnet pack Sources/Microcharts.Metapackage/Microcharts.Metapackage.csproj
+
+    - name: Add GitHub Packages source
+      run: nuget sources add -name github.com -source https://nuget.pkg.github.com/cabvision/index.json -username ${{ secrets.USERNAME }} -password ${{ secrets.NUGET_SOURCE_API_KEY }}
+
+    - name: Publish packages to nuget.org
```
Comment (prin53) [inline .github/workflows/ci.yml:51]: We are publishing to github.com, aren't we?

**Review by prin53 — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 4 above)
Comment (vladtaraban) [reply .github/workflows/ci.yml:50]: I took a look at other repos and saw that we use that source for NuGet packages. Should I change it?

Comment (prin53) [reply .github/workflows/ci.yml:50]: I mean that the step name is not correct: we are uploading NuGet packages to GitHub Package Manager, not to the nuget.org.

**Review by prin53 — APPROVED** [review]

**Pull request merged**